cpp/randomwalk_module/algorithm/random_walk.cpp

```diff
@@ -0,0 +1,86 @@
+#include <mg_utils.hpp>  // this line contains declarations of the public C API
+
+void RandomWalk(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result,
+                mgp_memory *memory);  // declaration of over our function
+
+
+void InsertStepRecord(mgp_graph *graph, mgp_result *result, mgp_memory *memory, const int step, const int node_id);
+
+
+extern "C" int mgp_init_module(
+    struct mgp_module *module,
+    struct mgp_memory *memory);  // by using extern, c++ compiler will not add argument/parameter type information to
+                                 // the name used for linkage
+// this is because C doesn't support function name overloading while C++ does
+// mgp_init_modules must be used so we can register procedures and in this way they can be called from openCypher
+
+extern "C" int mgp_shutdown_module() { return 0; }  // with this ypu can reset any global states or release some global resources
+
+// Important thing is that exceptions shouldn't cross the module boundary
+// don't allocate any global resource with memory argument - if you need to set up a global state, it is possible to do
+// it in the mgp_init_module
+
+void RandomWalk(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  try {
+    const auto start = mgp::value_get_vertex(
+        mgp::list_at(args, 0));  // idea is you have a list of arguments: 1. is starting node and second is length
+    const auto n_steps = mgp::value_get_int(mgp::list_at(args, 1));  // number of steps
+
+    srand(time(NULL));  // set random seed
+
+    const auto start_id = mgp::vertex_get_id(start).as_int;  // get id of a starting vertex
+    const auto graph = mg_utility::GetGraphView(
+        memgraph_graph, result, memory,
+        mg_graph::GraphType::kUndirectedGraph);  // do you have this functionality when writing Python code
```

Comment:
The Python API only provides directed graphs, but take a look at [`python/mage/node2vec/graph.py`](https://github.com/memgraph/mage/blob/main/python/mage/node2vec/graph.py) to see how undirected graphs have been done before.